anomaly/anomaly_detector.py

```
@@ -246,12 +248,15 @@ async def detect_anomaly(data: Dict) -> Tuple[bool, float]:
         if not _MODEL_LOADED:
             await load_model()
 
-        # Validate input
-        if not isinstance(data, dict):
+        # Validate input using TelemetryData
+        try:
+            validated_data = TelemetryData.validate(data)
```

Ruff (0.14.10)

Local variable `validated_data` is assigned to but never used

Remove assignment to unused variable `validated_data`

(F841)